File: tests/integration/features/file_traceability/_language_parsers/python/04_python_forward_function_in_class/file.py
class Foo:
    def hello_world(self):
        """
        @relation(REQ-1, scope=function)
        """
        # @relation(REQ-1, scope=range_start)
        print("hello world")  # noqa: T201
        # @relation(REQ-1, scope=line)
        print("hello world")  # noqa: T201
        print("hello world")  # noqa: T201
        # @relation(REQ-1, scope=range_end)

def hello_world_2():
    print("hello world")  # noqa: T201

File: tests/integration/features/file_traceability/_language_parsers/c/20_c_backward_and_forward_functions/file.c
// @relation(REQ-1, scope=file)

/**
 * @relation(REQ-2, scope=function, role=Implementation)
 * @relation(REQ-1, scope=function, role=Test)
 */
void foo() {
  // foo
  // comment comment comment comment comment comment comment comment comment comment comment
}

// @relation(REQ-3, scope=range_start)
void bar() {
  // bar
}
// @relation(REQ-3, scope=range_end)

void someFunc() {
  // someFunc
}

// @relation(REQ-3, scope=function)
void longFunctionName() {
  // boo
  // boo
  // boo
  // boo
  // @relation(REQ-4, scope=range_start)
  // boo
  // boo
  // boo
  // boo // @relation(REQ-1, scope=line)
  // boo
  // boo
  // @relation(REQ-4, scope=range_end)
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
  // boo
}
